src/log.c

```diff
@@ -167,10 +169,83 @@ static struct fi_ops_log ofi_default_log_ops = {
 	.log = ofi_log,
 };
 
+static int ofi_close_logging_fid(struct fid *fid)
+{
+	return 0;
+}
+
+static int ofi_bind_logging_fid(struct fid *fid, struct fid *bfid,
+				uint64_t flags);
+
+static struct fi_ops ofi_logging_ops = {
+	.size = sizeof(struct fi_ops),
+	.close = ofi_close_logging_fid,
+	.bind = ofi_bind_logging_fid,
+	.control = fi_no_control,
+	.ops_open = fi_no_ops_open,
+	.tostr = fi_no_tostr,
+	.ops_set = fi_no_ops_set,
+};
+
 static struct fid_logging log_fid = {
+	.fid = {
+		.fclass = FI_CLASS_LOG,
+		.ops = &ofi_logging_ops,
+	},
 	.ops = &ofi_default_log_ops,
 };
 
+static int ofi_close_import(struct fid *fid)
+{
+	/* Reset logging ops to default */
+	pthread_mutex_lock(&common_locks.ini_lock);
+	log_fid.ops = &ofi_default_log_ops;
```

Comment:
This doesn't quite work.  See comment below.  We need to reset each ops individually back to the default.

Reply:
Done